cli/src/config.rs

```diff
@@ -583,6 +584,58 @@ fn read_config_path(config_path: &Path) -> Result<config::Config, config::Config
         .build()
 }
 
+pub fn remove_config_value_from_file(
+    key: &ConfigNamePathBuf,
+    path: &Path,
+) -> Result<(), CommandError> {
+    // Read config
+    let config_toml = std::fs::read_to_string(path).or_else(|err| {
+        match err.kind() {
+            // If config doesn't exist yet, read as empty and we'll write one.
+            std::io::ErrorKind::NotFound => Ok("".to_string()),
+            _ => Err(user_error_with_message(
+                format!("Failed to read file {path}", path = path.display()),
+                err,
+            )),
+        }
+    })?;
+    let mut doc: toml_edit::Document = config_toml.parse().map_err(|err| {
+        user_error_with_message(
+            format!("Failed to parse file {path}", path = path.display()),
+            err,
+        )
+    })?;
+
+    let mut key_iter = key.components();
+    let last_key = key_iter.next_back().expect("key must not be empty");
+    let root: &mut dyn TableLike = doc.as_table_mut();
+    let target_table = key_iter.try_fold(root, |table, key| {
+        table
+            .get_mut(key)
+            .ok_or(config::ConfigError::NotFound(key.to_string()))
+            .and_then(|table| {
+                table
+                    .as_table_like_mut()
```

Comment:
nit: better to use `.as_table_mut()` for consistency. Think inline table as a single value.

Reply:
Good point, thanks!

Should we give people a hint in the error message if they try to unset a key of an inline table then?
Or is `"{key} is not a table"` good enough?

(Maybe I am overthinking this.)

Reply:
Great, happy to leave the `"{key} is not a table"`. In case people get confused, we can special-case it then.